x-pack/plugins/maps/public/shared/layers/styles/components/vector/vector_style_editor.js

```diff
@@ -43,52 +49,192 @@ export class VectorStyleEditor extends Component {
     }
   }
 
-  render() {
+  async _loadSupportedFeatures() {
+    const supportedFeatures = await this.props.layer.getSource().getSupportedFeatures();
+    const isPointsOnly = await this.props.loadIsPointsOnly();
+    const isLinesOnly = await this.props.loadIsLinesOnly();
+
+    let selectedFeature = VECTOR_FEATURE_TYPES.POLYGON;
```

Comment:
By auto-correcting the `selectedFeature`, clicking a feature-type that for a layer which does not have those shapes, will yields in a no-op. It looks as if that button is not working.

I would just not do this, and render the options anyway, even if that layer does not have that kind of data (users may want to edit that styling anyway, since they know they may have mixed feature-types at some point in that layer). 

On first load, I would select the `seletecdFeature` based on the actual data.

_maybe_ we could show a small help message when there is a mismatch between the shape-selection and the actual shapes in the data.

Reply:
This is a bug and has been resolved with https://github.com/elastic/kibana/pull/36280/commits/3e6d619b8c1840e64830e246f3a7b748caf2a577